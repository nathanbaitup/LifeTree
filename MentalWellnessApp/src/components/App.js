import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    return (
      <View style={styles.mainContainer}>
        <Text> Mental Wellness App </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
    mainContainer: {
        marginTop: 50,
    },
})
