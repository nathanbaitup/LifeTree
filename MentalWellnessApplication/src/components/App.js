import React, { Component } from 'react';
import { View, StyleSheet } from 'react-native';

import 'react-native-gesture-handler';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

// Importing the required third party modules to create the navigation bar.
import { NavigationContainer } from '@react-navigation/native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';

// Importing all of the different components for each different page.
import Home from './pages/Home';
import AddNewEntry from './pages/AddNewEntry';
import EntriesList from './pages/EntriesList';
import MoodJournal from './pages/MoodJournal';
import HRMonitoring from './pages/HRMonitoring';

// Variable that is useed to create all of the tabs required for the navigation bar.
const Tab = createBottomTabNavigator();

export default class App extends Component {
    render() {
        return (
            <View style={styles.mainContainer}>
              {/* The Navigation bar creation tag that sets the default tab to 'home', adds tint styling for when an
                option is selected and returns the user to the home tab if back is pressed from any tab other than home,
                where the application will close. 

                * REFERENCE ACCESSED 18/11/2021 https://aboutreact.com/react-native-bottom-navigation/
                * Used to learn how to create a navigation bar.
                */}
                <NavigationContainer>
                    <Tab.Navigator
                        initialRouteName='Home'
                        backBehavior='initialRoute'
                        screenOptions={{
                        'tabBarStyle': [
                          {
                            'display': 'flex',
                            'tabBarActiveTintColor': '#448aff',
                          },
                          null ]}} >

                        <Tab.Screen
                            name='View all Entries' // Adds the page heading to the top of the screen. Can be removed by removing line.
                            component={EntriesList} // The component to switch to when tab is pressed.
                            options={{
                                tabBarLabel: 'View all Entries',
                                tabBarIcon: ({ color, size }) => (
                                    <Icon name='book' color={color} size={size} /> // Sets the icon to the default color and size.
                                )}} />
                        <Tab.Screen
                            name='Add an Entry'
                            component={AddNewEntry}
                            options={{
                                tabBarLabel: 'Add an Entry',
                                tabBarIcon: ({ color, size }) => (
                                    <Icon name='plus-circle' color={color} size={size} />
                                )}} />
                        <Tab.Screen
                            name='Home' 
                            component={Home}
                            options={{
                                tabBarLabel: 'Home',
                                tabBarIcon: ({ color, size }) => (
                                    {/* Adds custom styling for center button to be bigger and circular. */},
                                    <View style={styles.centerNavBTN}>  
                                        <Icon name='home' color={color} size={size} />
                                    </View>
                                )}} />
                        <Tab.Screen
                            name='View Moods'
                            component={MoodJournal}
                            options={{
                                tabBarLabel: 'View Moods',
                                tabBarIcon: ({ color, size }) => (
                                    <Icon name='calendar-multiselect' color={color} size={size} />
                                )}} />
                        <Tab.Screen
                            name='Heart Rate Monitoring'
                            component={HRMonitoring}
                            options={{
                                tabBarLabel: 'HR Monitoring',
                                tabBarIcon: ({ color, size }) => (
                                    <Icon name='heart-pulse' color={color} size={size} />
                                )}} />

                    </Tab.Navigator>
                </NavigationContainer>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    mainContainer: {
        flex: 1,
        justifyContent: 'center',
    },
    centerNavBTN: {
        position: 'absolute',
        bottom: 5,
        height: 58,
        width: 58,
        borderRadius: 58,
        backgroundColor: '#00e676',
        justifyContent: 'center',
        alignItems: 'center',
    },
});
