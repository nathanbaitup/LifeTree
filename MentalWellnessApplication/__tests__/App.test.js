import 'react-native';
import React from 'react';
import App from '../src/components/App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Checks to see if the App renders', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
