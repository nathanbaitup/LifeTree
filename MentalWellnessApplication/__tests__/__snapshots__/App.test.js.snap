// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks if <App> renders correctly with and without user data Checks to see if the App renders 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessibilityIgnoresInvertColors={true}
                        style={
                          Object {
                            "height": "100%",
                            "opacity": 50,
                            "width": "100%",
                          }
                        }
                      >
                        <Image
                          source={
                            Object {
                              "testUri": "../../../src/resources/img/background.png",
                            }
                          }
                          style={
                            Array [
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              },
                              Object {
                                "height": "100%",
                                "width": "100%",
                              },
                              undefined,
                            ]
                          }
                        />
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "justifyContent": "center",
                              "paddingTop": 50,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "fontSize": 25,
                                "fontWeight": "bold",
                                "textAlign": "center",
                              }
                            }
                          >
                            Welcome to LifeTree
                          </Text>
                          <Text>
                            Log into your account below:
                          </Text>
                        </View>
                        <RCTScrollView>
                          <View>
                            <View
                              style={
                                Object {
                                  "backgroundColor": "#FFFFFF",
                                  "borderRadius": 25,
                                  "justifyContent": "center",
                                  "margin": 20,
                                  "padding": 10,
                                }
                              }
                            >
                              <Image
                                source={
                                  Object {
                                    "testUri": "../../../src/resources/img/faces/happy.png",
                                  }
                                }
                                style={
                                  Object {
                                    "alignSelf": "center",
                                    "height": 100,
                                    "marginBottom": 10,
                                    "marginTop": 20,
                                    "width": 134,
                                  }
                                }
                              />
                              <TextInput
                                autoCapitalize="none"
                                onChangeText={[Function]}
                                placeholder="enter your email address..."
                                placeholderTextColor="#aaaaaa"
                                style={
                                  Object {
                                    "backgroundColor": "#f0efed",
                                    "borderRadius": 5,
                                    "height": 50,
                                    "marginBottom": 10,
                                    "marginLeft": 20,
                                    "marginRight": 20,
                                    "marginTop": 10,
                                    "overflow": "hidden",
                                    "paddingLeft": 10,
                                  }
                                }
                                testID="emailInput"
                                value=""
                              />
                              <TextInput
                                autoCapitalize="none"
                                onChangeText={[Function]}
                                placeholder="enter your password..."
                                placeholderTextColor="#aaaaaa"
                                secureTextEntry={true}
                                style={
                                  Object {
                                    "backgroundColor": "#f0efed",
                                    "borderRadius": 5,
                                    "height": 50,
                                    "marginBottom": 10,
                                    "marginLeft": 20,
                                    "marginRight": 20,
                                    "marginTop": 10,
                                    "overflow": "hidden",
                                    "paddingLeft": 10,
                                  }
                                }
                                testID="passwordInput"
                                value=""
                              />
                              <View
                                accessibilityLabel="Log In button"
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                nativeID="animatedComponent"
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "center",
                                    "backgroundColor": "#00e676",
                                    "borderRadius": 15,
                                    "justifyContent": "center",
                                    "marginTop": 20,
                                    "opacity": 1,
                                    "padding": 15,
                                    "width": 150,
                                  }
                                }
                                testID="loginButton"
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#000000",
                                      "fontSize": 18,
                                      "fontWeight": "bold",
                                      "textAlign": "center",
                                    }
                                  }
                                >
                                   Log In 
                                </Text>
                              </View>
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "marginTop": 20,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "fontSize": 16,
                                    }
                                  }
                                >
                                  Don't have an account yet?
                                  <Text
                                    onPress={[Function]}
                                    style={
                                      Object {
                                        "color": "#6db8bf",
                                        "fontSize": 16,
                                        "fontWeight": "bold",
                                      }
                                    }
                                    testID="createAccountLink"
                                  >
                                     Sign up
                                  </Text>
                                   
                                </Text>
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
